qiita_client/plugin.py

```diff
@@ -0,0 +1,330 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+import traceback
+import sys
+from string import ascii_letters, digits
+from random import SystemRandom
+from os.path import exists, join, expanduser
+from os import makedirs, environ
+from future import standard_library
+from json import dumps
+
+from qiita_client import QiitaClient
+
+with standard_library.hooks():
+    from configparser import ConfigParser
+
+
+class QiitaCommand(object):
+    """A plugin command
+
+    Parameters
+    ----------
+    name : str
+        The command name
+    description : str
+        The command description
+    function : callable
+        The function that executes the command. Should be a callable that
+        conforms to the signature:
+            `function(qclient, job_id, job_parameters, output_dir)`
```

Comment:
No, they can't be nullable. Yup, there is a return signature, I've updated the documentation to note that.